Dockerfile-nb

```diff
@@ -0,0 +1,10 @@
+FROM open_discussions_python
```

Comment:
This would probably be better accomplished with a docker [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/) in `Dockerfile` since this is depending on a locally built. This and the primary stage can then be targeted by specifying [`target`](https://docs.docker.com/compose/compose-file/#target), although it would require upgrading our compose definition to 3.4. This would guarantee that `open_discussions_python` is actually built and up-to-date and this container gets rebuilt when the upstream one does.

Reply:
OK, tried this out and pushed the results to this branch: `initial_config_and_doc_updates_multistage`

2 things: 
- I don't want the `notebook` service to run any time you run `docker-compose up`, so the only option is to define the service in a separate file. Since it's in a separate file, I can't use the `x-environment` variables defined in `docker-compose.yml`. I'm sure there's a clean way to get around this instead of copying the vars like I did, but I'm guessing it will involve moving them out of the docker-compose file
- When I run the notebook from the multistage-built notebook container, the running notebooks never finish connecting to the kernel. Maybe a Dockerfile error but at this point I think I've spent more time than it's worth trying to figure this out (at least for today)

If anything jumps out at you, let me know